apps/src/aichat/redux/aichatRedux.ts

```diff
@@ -201,51 +198,85 @@ const saveAiCustomization = async (
+};
 
-  dispatch(setSavedAiCustomizations(trimmedCurrentAiCustomizations));
-
-  const changedProperties = findChangedProperties(
-    savedAiCustomizations,
-    trimmedCurrentAiCustomizations
-  );
-  if (
-    changedProperties.some(property =>
-      [
-        'selectedModelId',
-        'temperature',
-        'systemPrompt',
-        'retrievalContexts',
-      ].includes(property)
-    )
-  ) {
-    dispatch(setNewChatSession());
-  }
+// Thunk called after a save has completed successfully.
+// Updates the chat window and reports analytics as necessary.
+export const onSaveComplete =
+  () => (dispatch: AppDispatch, getState: () => RootState) => {
+    const {savedAiCustomizations, currentAiCustomizations, currentSaveType} =
+      getState().aichat;
 
-  changedProperties.forEach(property => {
-    dispatch(
-      addChatMessage({
-        id: getNewMessageId(),
-        role: Role.MODEL_UPDATE,
-        chatMessageText:
-          AI_CUSTOMIZATIONS_LABELS[property as keyof AiCustomizations],
-        status: Status.OK,
-        timestamp: getCurrentTime(),
-      })
+    const changedProperties = findChangedProperties(
+      savedAiCustomizations,
+      currentAiCustomizations
     );
-    if (eventDescription) {
-      analyticsReporter.sendEvent(
-        eventDescription,
-        {
-          propertyUpdated: property,
-          levelPath: window.location.pathname,
-        },
-        PLATFORMS.BOTH
+    if (
+      changedProperties.some(property =>
+        [
+          'selectedModelId',
+          'temperature',
+          'systemPrompt',
+          'retrievalContexts',
+        ].includes(property)
+      )
+    ) {
+      dispatch(setNewChatSession());
+    }
+
+    changedProperties.forEach(property => {
+      dispatch(
+        addChatMessage({
+          id: getNewMessageId(),
+          role: Role.MODEL_UPDATE,
+          chatMessageText:
+            AI_CUSTOMIZATIONS_LABELS[property as keyof AiCustomizations],
+          status: Status.OK,
+          timestamp: getCurrentTime(),
+        })
       );
+
+      if (currentSaveType) {
+        analyticsReporter.sendEvent(
+          saveTypeToAnalyticsEvent[currentSaveType],
+          {
+            propertyUpdated: property,
+            levelPath: window.location.pathname,
+          },
+          PLATFORMS.BOTH
+        );
+      }
+    });
+
+    // Update our last saved customizations to match the current customizations
+    dispatch(setSavedAiCustomizations(currentAiCustomizations));
+    // Notify the UI that the save is complete.
+    dispatch(endSave());
+    // Go to the presentation page if we just finished publishing the model card.
+    if (currentSaveType === 'publishModelCard') {
+      dispatch(setViewMode(ViewMode.PRESENTATION));
     }
-  });
+  };
+
+// Thunk called when a save has failed.
+export const onSaveFail = () => (dispatch: AppDispatch) => {
+  dispatch(
+    addChatMessage({
```

Comment:
Did you confirm that these don't show up in conversation history logs? I think they won't b/c no sessionId, but worth confirming.

Reply:
I confirmed they're not sent to backend.